### PR TITLE
feat: Align v1 and v2 schemas better

### PR DESCRIPTION
**Pull request by jgadling**
This PR tries to align the v1 and v2 api schemas better - reusing field names and types where possible, even if it makes maintaining our linkml schema a little bit less orderly. For example: we have `Deposition.deposition_publications` and `Dataset.dataset_publications` fields, when both `Deposition` and `Dataset` could inherit a `publications` field from a mixin - but we're preferring to keep backwards compatibility in this case.

I also tried to keep the descriptions for each field in sync with the values in apiv1 -- these descriptions ultimately wind up in the documentation for the API Client!

The main changes in this PR are in `schema/schema.yaml` with a few related changes to `db_import/**` and `scrape.py` to support field renaming.

### CODE REVIEW

File: apiv2/schema/schema.yaml [hunk 1]
```diff
@@ -699,25 +677,36 @@ classes:
...
     - AuthorEntityMixin
     attributes:
       annotation:
         name: annotation
-        description: Metadata about an annotation for a run
+        description: Reference to the annotation this author contributed to
         from_schema: cdp-dataset-config
         inverse: Annotation.authors
         range: Annotation
+    slot_usage:
```
Comment (jgadling) [inline apiv2/schema/schema.yaml:692]: We're inheriting a bunch of fields from the `AuthorEntityMixin` and this `slot_usage` block lets us override properties of the inherited fields

**Comment by jgadling** [comment]
NOTE: we should also handle for case where an annotation submitted by the dataset_author is the portal_standard.

**Comment by jgadling** [comment]
NOTE: need to add support for a `simulated` annotation method

**Comment by jgadling** [comment]
> NOTE: need to add support for a `simulated` annotation method

File: apiv2/database/models/dataset.py [hunk 2]
```diff
@@ -81,8 +81,8 @@ class Dataset(Base):
     deposition_date: Mapped[datetime.datetime] = mapped_column(DateTime(timezone=True), nullable=False)
     release_date: Mapped[datetime.datetime] = mapped_column(DateTime(timezone=True), nullable=False)
     last_modified_date: Mapped[datetime.datetime] = mapped_column(DateTime(timezone=True), nullable=False)
-    publications: Mapped[str] = mapped_column(String, nullable=True)
+    dataset_publications: Mapped[str] = mapped_column(String, nullable=True)
```
Comment (manasaV3) [inline apiv2/database/models/dataset.py:84]: What do you think of having this field be `publications` across the dataset, deposition and tomogram table as the prefix is redundant? 
We could use the `CrossReferenceMixin` to keep this field and the `related_database_entries` consistent.

File: apiv2/database/models/tomogram.py [hunk 3]
```diff
@@ -72,9 +72,9 @@ class Tomogram(Base):
         back_populates="tomograms",
     )
     name: Mapped[str] = mapped_column(String, nullable=True)
-    size_x: Mapped[float] = mapped_column(Float, nullable=False)
-    size_y: Mapped[float] = mapped_column(Float, nullable=False)
-    size_z: Mapped[float] = mapped_column(Float, nullable=False)
```
Comment (manasaV3) [inline apiv2/database/models/tomogram.py:77]: We should add `is_visualization_default`, `publications` and `related_database_entries` fields to this table as well.

File: apiv2/schema/schema.yaml [hunk 4]
```diff
@@ -816,7 +809,7 @@ classes:
       plural:
         tag: plural
         value: Annotations
-    description: Metadata about an annotation for a run
```
Comment (manasaV3) [inline apiv2/schema/schema.yaml:819]: I think this table is missing the `method_links` json field.

File: apiv2/database/models/tomogram.py (hunk 3 above)
Comment (manasaV3) [reply apiv2/database/models/tomogram.py:77]: Resolving it here, as I added this [comment to the other PR.](https://github.com/chanzuckerberg/cryoet-data-portal-backend/pull/280#discussion_r1774575521)

File: apiv2/database/models/dataset.py (hunk 2 above)
Comment (jgadling) [reply apiv2/database/models/dataset.py:84]: Basically this PR is specifically to try to keep as much compatibility as possible with the V1 api, which was the preference that CZII had for the migration, and in the old api, the field was `dataset_publications`

File: apiv2/schema/schema.yaml (hunk 4 above)
Comment (jgadling) [reply apiv2/schema/schema.yaml:819]: Added!

File: apiv2/schema/schema.yaml [hunk 5]
```diff
@@ -887,6 +879,9 @@ classes:
         - annotation_method
         range: string
         required: true
+      method_links:
+        description: Provides links to information on the method used for generating annotation, comma separated
+        range: string
```
Comment (manasaV3) [inline apiv2/schema/schema.yaml:884]: I believe this is currently a json field.

**Review by manasaV3 — APPROVED** [review]

**Pull request merged**